chainer/_backprop_utils.py

```diff
@@ -139,32 +139,34 @@ def backprop_step(
             raise ValueError(func._get_error_message(msg))
 
     for i, gx in six.moves.zip(target_input_indexes, gxs):
-        if gx is not None:
-            grad_inputs[func.inputs[i]].append(gx)
+        if gx is None or gx.array is None:
```

Comment:
It seems that the change will make Chainer more tolerant of returning `Variable(None)`.  I'd rather not treat `Variable(None)` as `None`.

Reply:
Currently Chainer does allow `Variable(None)`, so I think this check is legitimate.
Do you have any alternative?

Reply:
To preserve the behavior for now,
```python
        if gx is None:
            continue
        grad_inputs[func.inputs[i]].append(gx)
        if is_debug and gx.array is None:
            raise ...
```

Reply:
Do you mean we're going to disallow `Variable(None)` in this PR?

Reply:
I mean that `Variable(None)` is checked if `is_debug` before the PR.  I've not checked older codes.

Reply:
Where is the check? I cannot find it.
Currently some function implementations return `Variable(None)`, so that would mean they're broken in debug mode.

Reply:
OK.  No "check" for `Variable(None)` is there but a line with `gx.shape` just fails in debug mode.

As we don't have a consensus on `Variable(None)`, I'd like code-fix/performance PRs to preserve the behavior.

Reply:
Currently Chainer does (or should) support `Variable(None)`, because some built-in functions return such variables.
So I'd say the current behavior is a bug.
Perhaps we can put `cat:bug` label to this PR.

Reply:
The PR turns `Variable(None)` into `None`.

> Currently Chainer does (or should) support `Variable(None)`, because some built-in functions return such variables.
> So I'd say the current behavior is a bug.

Then the fix for the bug could be
```python
        if gx is None:
            continue
        grad_inputs[func.inputs[i]].append(gx)
        if gx.array is None:
            continue
```

Reply:
Does that make difference?
I was not sure about that point but I thought it should not, so I fixed as the current implementation.